### PR TITLE
[Python API] add new api 

### PR DESCRIPTION
**Pull request by akuporos**
### Details:
 - bind Core class:
   - available_devices
   - 	query_model()
   -   unload_plugin()
   -   register_plugins()
   -   register_plugin()
   -   get_metric()
   -   get_config()
   -   import_model()
   -   set_config()
   -   compile_model()
   -   get_versions()
   -   read_model()
 - bind ExecutableNetwork class
   -    infer_new_request()
   -  	get_config()
   -   get_metric()
   -   create_infer_request()
   -   get_runtime_function()
   -   export_model()
   -   output()
   -   outputs
   -   input()
   -   inputs 
 - bind InferRequest class
   -   input_tensors
   -   output_tensors
   -   latency
   -   profiling_info
   -   set_output_tensor()
   -   set_input_tensor()
   -   set_tensor()
   -   get_tensor()
   -   set_tensors()
   -   set_output_tensors()
   -   set_input_tensors()
   -   infer()
   -   start_async()
   -   wait()
   -   wait_for()
   -   cancel()
   -   set_callback()
 - added bindging to Function class
   -   reshape()
   -   input()
   -   inputs
   -   output()
   -   outputs()
 - bind const Node Outputs class
 - PR for FE: https://github.com/openvinotoolkit/openvino/pull/8301
### Tickets:
 - *ticket-id*

### CODE REVIEW

File: runtime/bindings/python/src/pyopenvino/core/core.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+    cls.def(
+        "get_metric",
+        [](ov::runtime::Core& self, const std::string device_name, const std::string name) -> py::handle {
+            return Common::parse_parameter(self.get_metric(device_name, name));
+        },
+        py::arg("device_name"),
+        py::arg("name"));
+
+    cls.def("register_plugin", &ov::runtime::Core::register_plugin, py::arg("plugin_name"), py::arg("device_name"));
```
Comment (ilyachur) [inline runtime/bindings/python/src/pyopenvino/core/core.cpp:89]: Could you align API with C++?
I don't see some methods, for example `add_extension`.

File: runtime/bindings/python/src/pyopenvino/core/ie_infer_queue.cpp [hunk 2]
```diff
@@ -145,89 +140,89 @@ class InferQueue {
...
+
+//     cls.def("get_idle_request_info", [](InferQueue& self) {
+//         return self._getIdleRequestInfo();
+//     });
+
+//     cls.def("set_infer_callback", [](InferQueue& self, py::function f_callback) {
+//         self.setCustomCallbacks(f_callback);
+//     });
+
+//     cls.def("__len__", [](InferQueue& self) {
```
Comment (ilyachur) [inline runtime/bindings/python/src/pyopenvino/core/ie_infer_queue.cpp:210]: Maybe we can remove this commented code

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+        py::arg("tensor"));
+
+    cls.def(
+        "set_output_tensor",
+        [](InferRequestWrapper& self, const ov::runtime::Tensor& tensor) {
+            self._request.set_output_tensor(tensor);
+        },
+        py::arg("tensor"));
+
+    cls.def_property_readonly("inputs", [](InferRequestWrapper& self) {
```
Comment (ilyachur) [inline runtime/bindings/python/src/pyopenvino/core/infer_request.cpp:257]: I don't see such API in C++

File: runtime/bindings/python/src/pyopenvino/core/ie_infer_queue.cpp (hunk 2 above)
Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/core/ie_infer_queue.cpp:210]: We'll adapt this code in the next PR, so decided to temporarily comment it

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp (hunk 3 above)
Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/core/infer_request.cpp:261]: @jiwaszki ping here

File: runtime/bindings/python/src/pyopenvino/core/core.cpp (hunk 1 above)
Comment (akuporos) [reply runtime/bindings/python/src/pyopenvino/core/core.cpp:89]: I added ```add_extension``` which loads extension from path. Other overloading will be added in a separate PR

File: runtime/bindings/python/src/pyopenvino/core/infer_request.cpp (hunk 3 above)
Comment (jiwaszki) [reply runtime/bindings/python/src/pyopenvino/core/infer_request.cpp:261]: @ilyachur We aligned as much as we were able to, sometimes there are some limits on Python side, sometimes due to on-going development of new C++ API.
However I know that we have some independence for Python API development - this method is just a helper to simplify access for the Python users. One of the example is inside tests runtime code (here for outputs):
https://github.com/openvinotoolkit/openvino/blob/3621e39f293d2daf3cf65a8a1bdb901959dce11a/runtime/bindings/python/tests/runtime.py#L138
I do not see how it affects alignment in a wrong way, it only provides user-friendly way to access data.

File: inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp [hunk 4]
```diff
@@ -305,7 +305,7 @@ InferenceEngine::Blob::Ptr MKLDNNPlugin::MKLDNNInferRequest::GetBlob(const std::
                                                                                                          desc.getShape().getRank()))
                                                                    : MemoryDescUtils::convertToTensorDesc(desc);
                         const auto &tensorDesc = data->getTensorDesc();
-                        if (expectedTensorDesc.getPrecision() != tensorDesc.getPrecision()) {
+                        if (expectedTensorDesc.getPrecision() != normalizeToSupportedPrecision(tensorDesc.getPrecision())) {
```
Comment (dmitry-gorokhov) [inline inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:308]: @jiwaszki Could you clarify which tests or use-cases require this change?

**Review by dmitry-gorokhov — APPROVED** [review]

File: inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp (hunk 4 above)
Comment (akuporos) [reply inference-engine/src/mkldnn_plugin/mkldnn_infer_request.cpp:308]: This one test fails without this change: ```tests/test_onnx/test_zoo_models.py::OnnxBackendModelExecutionTest::test_onnx_model_zoo_text_machine_comprehension_bert_squad_model_bertsquad_10_bertsquad_10_bertsquad10_cpu```
https://dev.azure.com/openvinoci/dldt/_build/results?buildId=197937&view=logs&j=8e6fb299-7deb-52f1-2f4a-df91230aba00&t=f812e137-0344-51cb-e8b0-0ab752070263&l=6178

**Pull request merged**